server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -230,4 +207,77 @@ protected void checkDuplicates(
     updatedDuplicateOptions.putAll(customDuplicateOptions);
     checkDuplicateEntries(ctx, foundEntries, updatedDuplicateOptions);
   }
+
+  /**
+   * Flags errors for rule lists passed as parameters if there are multiple instances of mutually
+   * exclusive options.
+   *
+   * @param options Options checked to insert into error message
+   * @param rules Lists of TerminalNode to iterate through
+   * @return Number of TerminalNode instances found
+   */
+  protected int checkHasMutuallyExclusiveOptions(String options, List<TerminalNode>... rules) {
```

Comment:
No generic and check for type? What about if we want to enforce sets of ParserRuleContext variables inconjunction with TerminalNodes as well? 

I had the best of both worlds set up, but the PRs are going slowly. See my commit here: https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/compare/development...mm-broadcom:che-che4z-lsp-for-cobol:CICS-WEB-Command-Fix-V2#diff-50eac5e66277f894df3f0b69a3d008dfc1aa9d9f0460c5c3c1b5ff41898af1f7R203